src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -41,6 +41,20 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * Parses {@code zeroBasedIndex} into an {@code Index} and returns it. Leading and trailing whitespaces will be
+     * trimmed.
+     *
+     * @throws ParseException if the specified index is invalid (not non-zero unsigned integer).
+     */
+    public static Index parseLinkIndex(String zeroBasedIndex) throws ParseException {
+        String trimmedIndex = zeroBasedIndex.trim();
```

Comment:
Based on the debugger, zeroBasedIndex is null since
```Index linkIndex = ParserUtil.parseLinkIndex(Objects.requireNonNull(argMultimap
                .getValue(PREFIX_LINK_INDEX)).orElse(null));
```
orElse(null) is being used for the value, so trim() throws a NullPointerException
```suggestion
        if (zeroBasedIndex == null) {
            return null;
        }
        String trimmedIndex = zeroBasedIndex.trim();
```